File: task.py
from hashlib import sha1
import hmac
import httplib
import json
import pprint
import re

pp = pprint.PrettyPrinter(indent=4)

host   = 'api.groupcamp.com'
key    = "token from the API icon in the admin panel of your GroupCamp account"
secret = "the secret associated with the above key"
debug  = 1
acting = 'email address of the "acting" user (not that important for the reporting API)'


def do_call (method, path, body='', q_string='') :

	# The text to be signed is a concatenation of
	# - the method (POST, PUT, GET, etc),
	# - the string '::'
	# - the path, with the leading /, without the trailing /
	# - the string '::'
	# - the body of the request (can be a json object, or an empty string)

	text = method + '::' + path + '::' + body
	signature = hmac.new(secret, text, sha1)
	sig = signature.digest().encode("base64").rstrip('\n')

	# X-Gcmp-Application MUST be 'reporting-1'
	# X-Gcmp-Acting MUST be set to the user which will be logged as performing the action.
	#   You can create a "fake" user in your GroupCamp account for this purpose. User
	#   privileges are not checked - this user is *only* logged as the user who did
	#   perform the action.
	# Authorization is the string "GCMP <key>:<signature>", the secret MUST NEVER BE SENT.
	#   It should be kept secret.
	# Content-Type is application/json, since the body is a json object.

	headers = {
		'X-Gcmp-Application': 'reporting-1',
		'X-Gcmp-Acting': acting,
		'Authorization': 'GCMP '+key+':'+sig,
		'Content-Type': 'application/json'
	}

	if debug :
		print "The text to be signed <"+text+'>'
		print "The obtained signature "+sig

	if method == 'GET' and q_string != '':
		path = path + '?' + q_string

	cnx = httplib.HTTPSConnection(host, 443)
	cnx.request(method, path, body, headers)
	res = cnx.getresponse()
	return res



def get_element(method, path) :
	res = do_call(method, path)
	status = res.status
	body = res.read()

	if debug :
		print "Status ",status
		print "Raw answer ",body

	data = json.loads(body)

	if status != 200 :
		print "Failed"
		return 'failed'
	else :
		print "Request successful"
		result = data['result']
	return result


result = get_element('GET', '/task/v1/task/[UUID of a task]')
pp.pprint(result)

File: task.js
/*
	This tiny test script is designed to be used in a browser.

	Include those in the HEAD of your HTML page:
	<script src="https://cdnjs.cloudflare.com/ajax/libs/crypto-js/3.1.9-1/crypto-js.min.js"></script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/crypto-js/3.1.9-1/hmac-sha1.min.js"></script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/crypto-js/3.1.9-1/enc-base64.min.js"></script>
 */

// This one is the key, declared for your API access in the GroupCamp admin interface.
var key    = 'The key';
// This one is the secret associated with the key, used to sign your request. Must not
// be sent to GroupCamp (or elsewhere).
var secret = 'The secret associated to the key';
// The main GroupCamp API server
var domain = 'api.groupcamp.com';
// The user acting
var acting = 'e-mail address of the acting user';

/*
	Will send a request to the GroupCamp API, and display the result in the web page.
*/
function send_request(method, path, body, q_string, on_success, on_error) {
	// Signing request
	var text = method + '::' + path + '::' + body
	var hash = CryptoJS.HmacSHA1(text, secret);
	var hashInBase64 = CryptoJS.enc.Base64.stringify(hash);

	// Sending request
	var url = 'https://'+domain+path;
	if ( q_string ) {
		url += '?'+q_string; 
	} 
	var xhr = new XMLHttpRequest();
	xhr.open(method, url, true);
	xhr.onload = function(res) { on_success(xhr.response); };
	xhr.onerror = on_error;
	xhr.withCredentials = false;
	xhr.responseType = 'json';
	xhr.setRequestHeader('Content-Type','application/json');
	xhr.setRequestHeader('X-Gcmp-Application','reporting-1');
	xhr.setRequestHeader('X-Gcmp-Acting',acting);
	xhr.setRequestHeader('Authorization','GCMP '+key+':'+hashInBase64);
	xhr.send(body);
}

function get_task(uuid, on_success, on_error) {
	var method = 'GET';
	var path = '/task/v1/task/' + uuid;
	var body = '';
	var q_string = ''
	send_request(method, path, body, q_string, on_success, on_error);
}

var uuid = 'Uuid of a task (you can get it from the information panel)';
get_task(uuid, function(res) {console.info(res);}, function(res) {console.error(res);});

